.github/workflows/lint_python.yml

```diff
@@ -0,0 +1,24 @@
+name: lint_python
+on: [pull_request, push]
+jobs:
+  lint_python:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - run: pip install bandit black codespell flake8 isort mypy pytest pyupgrade safety
+      - run: bandit --recursive --skip B101 . || true  # B101 is assert statements
+      - run: black --check . || true
+      - run: codespell  # --ignore-words-list="" --skip=""
+      - run: flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+      - run: isort --check-only --profile black . || true
+      - run: pip install tox -r test-requirements.txt
+      - run: pip install -e .
+      # - run: GS_CREDS_FILENAME=<YOUR_CREDS.json> make test
```

Comment:
I ran some scenarios, I found a way to run the tests in the CI, I will need to update the files under `tests/cassettes/` then this should pass. :rocket: 

*Important:* we should not provide a value in `GS_CREDS_FILENAME` if we wish to run i in the CI, one should only do so to run the tests using a real HTTP client and make each requests. :teacher: 

I'll advise to remove that comment, ro prevent confusion :slightly_smiling_face:

Reply:
GitHub Actions can take secrets but removed anyways.

Reply:
it is not a matter of storing secrets it's the way the test suite is configured. if credentials are provided then it will try to connect to the internet and make the real request, if not provided and the default record mode is set to `once` or `none`then no actual connection to a real spreadsheet is made.

Reply:
Removed anyways.